Framework/API/src/ISISInstrumentDataCache.cpp

```diff
@@ -45,8 +41,40 @@ std::string Mantid::API::ISISInstrumentDataCache::getFileParentDirectoryPath(con
   return dirPath;
 }
 
+/**
+ * Get a vector of the run numbers' files present in the given instrument's data cache.
+ * @param instrumentName The instrument to get run numbers for.
+ * @return A vector containing the run numbers that can be accessed from the data cache.
+ */
+std::vector<std::string> ISISInstrumentDataCache::getRunNumbersInCache(const std::string &instrumentName) const {
+  const auto &json = openCacheJsonFile(instrumentName).second;
+  return json.getMemberNames();
+}
+
+/**
+ * Open the Json file and return the path and the opened file object.
+ * @param instrumentName The name of the instrument to open the index file for.
+ * @return A pair containing the path and the contents of the json file.
+ * @throws std::invalid_argument if the file could not be found.
+ */
+std::pair<std::string, Json::Value>
+ISISInstrumentDataCache::openCacheJsonFile(std::string const &instrumentName) const {
+  // Open index json file
+  std::filesystem::path const &jsonPath =
+      std::filesystem::path(m_dataCachePath) / instrumentName / (instrumentName + "_index.json");
+  std::ifstream ifstrm{jsonPath};
+  if (!ifstrm) {
+    throw std::invalid_argument("Could not open index file: " + jsonPath.string());
+  }
```

Comment:
It looks like the old code attempted to use the short name of the instrument if the full name didn't work, does it not need to do that any more?

Reply:
Ah, I think (very much not certain) that this was added later and so got missed during a rebase. Good catch, thanks